CONTRIBUTING.md

```diff
@@ -0,0 +1,134 @@
+# Contributing to Flare
+
+First off, thank you for considering contributing! 🎉  
+We welcome contributions from everyone — bug reports, feature requests, documentation improvements, or code enhancements.
+
+---
+
+## Table of Contents
+- [Code of Conduct](#code-of-conduct)
+- [How to Contribute](#how-to-contribute)
+- [Development Setup](#development-setup)
+- [Branching & Commits](#branching--commits)
+- [Pull Requests](#pull-requests)
+- [Style Guide](#style-guide)
+- [Reporting Bugs](#reporting-bugs)
+- [Feature Requests](#feature-requests)
+
+---
+
+## Code of Conduct
+By participating in this project, you agree to follow our [Code of Conduct](./CODE_OF_CONDUCT.md).  
+Be respectful, constructive, and welcoming to all contributors.
+
+---
+
+## How to Contribute
+You can contribute in many ways:
+- **Bug reports:** Open an issue with steps to reproduce.
+- **Feature requests:** Suggest new features or enhancements.
+- **Documentation:** Improve README, examples, or type definitions.
+- **Code contributions:** Add features, fix bugs, or refactor code.
+
+---
+
+## Development Setup
+1. **Fork the repo**  
+2. **Clone your fork locally:**
+   ```bash
+   git clone https://github.com/xafans/flare.git
+   cd flare
+   ```
+
+3. **Install dependencies:**
+
+   ```bash
+   npm install
+   ```
+4. **Build the project:**
+
+   ```bash
+   npm run build
+   ```
+5. **Run tests:**
+
+   ```bash
+   npm test
+   ```
+6. **Make changes** in a feature branch (see Branching & Commits below).
+
+---
+
+## Branching & Commits
+
+* Create a **feature branch** from `main`:
+
+  ```bash
+  git checkout -b chore/your-feature
+  ```
+* Use **kebab-case** for branch names.
+* Use **conventional commits**:
```

Comment:
remove this section since we are using squash merge and will use our own commit messages eventually